libmachine/provision/suse.go

```diff
@@ -205,3 +182,34 @@ func (provisioner *SUSEProvisioner) Provision(swarmOptions swarm.Options, authOp
 	err = provisioner.Service("docker", serviceaction.Enable)
 	return err
 }
+
+func (provisioner *SUSEProvisioner) configureFirewall() error {
+	// Update cloud-netconfig and restart the machine
+	if _, err := provisioner.SSHCommand("sudo -E zypper -q -n update cloud-netconfig-ec2"); err != nil {
+		return err
+	}
+	provisioner.SSHCommand("sudo shutdown -r now")
+	if err := drivers.WaitForSSH(provisioner.Driver); err != nil {
+		return err
+	}
```

Comment:
Although it seems like the community AMI referenced in https://github.com/rancher/rancher/issues/25071 seems to require this step to provision correctly, this does not seem to be the case in certain other AMIs, such as the following "Quick Start" amis in us-east-2:

`SUSE Linux Enterprise Server 15 SP1 (HVM), SSD Volume Type - ami-013d888fbcf1a3962`
`SUSE Linux Enterprise Server 12 SP5 (HVM), SSD Volume Type - ami-0185282502c4b22be`

Furthermore, it seems to introduce a flaky provisioning bug as a result of the zypper update, seen in the following screenshot, most commonly in SLES 15 SP 1:
<img width="1311" alt="Screen Shot 2020-05-07 at 1 26 17 AM" src="https://user-images.githubusercontent.com/13127443/81274012-93abcc00-9004-11ea-9dd8-e82489e2e876.png">

Since this introduces cloud-provider-specific logic into the provisioning step (as it seems ec2 specific), should this step be omitted from the process in favor of requiring users to use AMIs that are already updated or should this update be enforced from the provisioning side?

Reply:
Double-checked on Azure and we're able to provision SUSE nodes perfectly fine without:
```
// Update cloud-netconfig and restart the machine
if _, err := provisioner.SSHCommand("sudo -E zypper -q -n update cloud-netconfig-ec2"); err != nil {
	return err
}
provisioner.SSHCommand("sudo shutdown -r now")
if err := drivers.WaitForSSH(provisioner.Driver); err != nil {
	return err
}
```

Since this seems to be AMI-specific, I'm removing this check in the latest commit and making a note on the ticket (https://github.com/rancher/rancher/issues/25071).